### PR TITLE
Implement the `node` C# function

### PR DESCRIPTION
**Pull request by Porges**
## Summary of the Pull Request

1. Ports the `node` function from Python to C#.
2. Adds a missing authentication check.
3. Add validated string type `PoolName` for consistency with Python version.

## PR Checklist
* [x] Applies to work item: #2068 
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/onefuzz) and sign the CLI.
* [x] Tests added/passed
* ~~Requires documentation to be updated~~
* ~~I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx~~

## Info on Pull Request

_What does this include?_

## Validation Steps Performed

_How does someone test & validate?_

### CODE REVIEW

File: src/ApiService/ApiService/OneFuzzTypes/Model.cs [hunk 1]
```diff
@@ -316,11 +314,11 @@ public record InstanceConfig
...
-    IDictionary<Endpoint, ApiAccessRule>? ApiAccessRules,
-    IDictionary<PrincipalId, GroupId[]>? GroupMembership,
-
-    IDictionary<string, string>? VmTags,
-    IDictionary<string, string>? VmssTags
+    IDictionary<Endpoint, ApiAccessRule>? ApiAccessRules = null,
+    IDictionary<PrincipalId, GroupId[]>? GroupMembership = null,
+    IDictionary<string, string>? VmTags = null,
+    IDictionary<string, string>? VmssTags = null,
+    bool? RequireAdminPrivileges = null
```
Comment (Porges) [inline src/ApiService/ApiService/OneFuzzTypes/Model.cs:321]: This was left out of the original C# version. Are we looking to remove/deprecate this functionality?

Comment (tevoinea) [reply src/ApiService/ApiService/OneFuzzTypes/Model.cs:321]: It's possible it was added to the python side after the C# version was created

File: src/ApiService/ApiService/OneFuzzTypes/Responses.cs [hunk 2]
```diff
@@ -15,6 +18,23 @@ public record PendingNodeCommand(
     NodeCommandEnvelope? Envelope
 ) : BaseResponse();
 
+// TODO: not sure how much of this is actually
+// needed in the search results, so at the moment
+// it is a copy of the whole Node type
```
Comment (tevoinea) [inline src/ApiService/ApiService/OneFuzzTypes/Responses.cs:23]: Could we turn `BaseResponse` into an interface, add it on the existing `Node` record in `Model.cs`? The last thing to figure out would be the BaseResponse JsonConverter

File: src/ApiService/ApiService/onefuzzlib/orm/Queries.cs [hunk 3]
```diff
@@ -11,6 +11,9 @@ public static class Query {
         public static string PartitionKey(string partitionKey)
             => TableClient.CreateQueryFilter($"PartitionKey eq {partitionKey}");
 
+        public static string PartitionKey(Guid partitionKey)
+            => TableClient.CreateQueryFilter($"PartitionKey eq {partitionKey}");
+
```
Comment (tevoinea) [inline src/ApiService/ApiService/onefuzzlib/orm/Queries.cs:16]: What about something like:
```csharp
public static string PartitionKey<T>(T partitionKey)
            where T : IFormattable
            => TableClient.CreateQueryFilter($"PartitionKey eq {partitionKey}");
```

That way we can also add it to `PoolName` for example.

**Review by tevoinea — APPROVED** [review]

File: src/ApiService/ApiService/OneFuzzTypes/Responses.cs (hunk 2 above)
Comment (Porges) [reply src/ApiService/ApiService/OneFuzzTypes/Responses.cs:23]: I think there's probably going to be things in `Node` (now or in future) that we don't want to reflect into the response here?

File: src/ApiService/ApiService/onefuzzlib/orm/Queries.cs (hunk 3 above)
Comment (Porges) [reply src/ApiService/ApiService/onefuzzlib/orm/Queries.cs:16]: Also had this idea but hadn’t gotten around to it 👍

File: src/ApiService/ApiService/OneFuzzTypes/Responses.cs (hunk 2 above)
Comment (Porges) [reply src/ApiService/ApiService/OneFuzzTypes/Responses.cs:23]: I’ve also been bitten a lot of times in the past by sharing internal models as external ones, so I think I'll leave it like this for now.

**Pull request merged**